### PR TITLE
feat/ documentation technical + user

### LINKED ISSUE #83: Feat documentation (technical + user)

### PR DESCRIPTION
**Pull request by theophile-jr**
Features : 
* Technical doc pdf
* User doc pdf
* refman.pdf : automatically generated documentation by doxygen with "make doc"

### CODE REVIEW

File: doc/.gitkeep
Comment (Njord201) [inline doc/.gitkeep:1]: Remove this file.

**Review by Njord201 — CHANGES_REQUESTED** [review]

File: include/Parser/Factory.hpp [hunk 1]
```diff
@@ -49,24 +49,32 @@ class Raytracer::Factory {
          */
         ~Factory() = default;
 
-        /// @brief Create a type component
-        /// @param type Component to create
-        /// @return The Component
+        /*
+         * @brief Create a type component
```
Comment (AxelF44) [inline include/Parser/Factory.hpp:53]:
```suggestion
         * @brief Create a type component.
```

File: include/Parser/Factory.hpp [hunk 2]
```diff
@@ -49,24 +49,32 @@ class Raytracer::Factory {
...
+         * @param type Component to create
+         * @return The Component
+         */
         std::shared_ptr<Primitive::IPrimitive> createPrimitivesComponent(const std::string &type);
 
-        /// @brief Register a type component
-        /// @param type Component to register
-        /// @param creator Function to create the component
+        /*
+         * @brief Register a type component
```
Comment (AxelF44) [inline include/Parser/Factory.hpp:60]:
```suggestion
         * @brief Register a type component.
```

File: include/Parser/Factory.hpp [hunk 3]
```diff
@@ -49,24 +49,32 @@ class Raytracer::Factory {
...
+         * @param type Component to register
+         * @param creator Function to create the component
+         */
         void registerPrimitivesComponent(const std::string& type, PrimitivesCreator creator);
 
-        /// @brief Create a type component
-        /// @param type Component to create
-        /// @return The Component
+        /*
+         * @brief Create a type component
```
Comment (AxelF44) [inline include/Parser/Factory.hpp:67]:
```suggestion
         * @brief Create a type component.
```

File: include/Parser/Factory.hpp [hunk 4]
```diff
@@ -49,24 +49,32 @@ class Raytracer::Factory {
...
+         * @param type Component to create
+         * @return The Component
+         */
         std::shared_ptr<Light::ILight> createLightsComponent(const std::string &type);
 
-        /// @brief Register a type component
-        /// @param type Component to register
-        /// @param creator Function to create the component
+        /*
+         * @brief Register a type component
```
Comment (AxelF44) [inline include/Parser/Factory.hpp:74]:
```suggestion
         * @brief Register a type component.
```

File: include/Parser/Scene.hpp [hunk 5]
```diff
@@ -69,12 +69,15 @@ class Raytracer::Scene {
 
         class ParserException : public std::exception {
             public:
-                /// @brief Exception to throw when there is an error in parsing
-                /// @param msg message to throw
+                /*
+                 * @brief Exception to throw when there is an error in parsing
```
Comment (AxelF44) [inline include/Parser/Scene.hpp:73]:
```suggestion
                 * @brief Exception to throw when there is an error in parsing.
```

File: include/Parser/Scene.hpp [hunk 6]
```diff
@@ -69,12 +69,15 @@ class Raytracer::Scene {
...
+                 * @param msg message to throw
+                 */ 
                 explicit ParserException(const std::string &msg) : message(msg) {}
                 
-                /// @brief Return the message of error
-                /// @return message
+                /* @brief Return the message of error
```
Comment (AxelF44) [inline include/Parser/Scene.hpp:78]:
```suggestion
                /* @brief Return the message of error.
```

File: include/Parser/Scene.hpp [hunk 7]
```diff
@@ -84,36 +87,47 @@ class Raytracer::Scene {
...
-        /// @param config of the current file
-        /// @return integer 0 if no error else x
-        int _parseScenesImports(const libconfig::Config &config);
-
-        /// @brief Parse the imported scene
-        /// @param filePath of the scene to parse
-        /// @return integer 0 if no error else x
-        int parseImportedScene(std::string filePath);
+            /*
+             * @brief Parse the Camera settings
```
Comment (AxelF44) [inline include/Parser/Scene.hpp:91]:
```suggestion
             * @brief Parse the Camera settings.
```

File: include/Parser/Scene.hpp [hunk 8]
```diff
@@ -84,36 +87,47 @@ class Raytracer::Scene {
...
+             * @param camera Settings
+             * @return 0 if no error else 84
+             */
+            int _parseCameraSetting(const libconfig::Setting &camera);
+
+            /*
+             * @brief Parse the Primitives settings
```
Comment (AxelF44) [inline include/Parser/Scene.hpp:98]:
```suggestion
             * @brief Parse the Primitives settings.
```

File: include/Parser/Scene.hpp [hunk 9]
```diff
@@ -84,36 +87,47 @@ class Raytracer::Scene {
...
+             * @param camera Settings
+             * @return 0 if no error else 84
+             */
+            int _parsePrimitiveSetting(const libconfig::Setting &primitives);
+
+            /*
+             * @brief Parse the Lights settings
```
Comment (AxelF44) [inline include/Parser/Scene.hpp:105]:
```suggestion
             * @brief Parse the Lights settings.
```

File: include/Parser/Scene.hpp [hunk 10]
```diff
@@ -84,36 +87,47 @@ class Raytracer::Scene {
...
+             * @param camera Settings
+             * @return 0 if no error else 84
+             */
+            int _parseLightsSetting(const libconfig::Setting &lights);
+
+            /*
+             * @brief Parse the value of settings
```
Comment (AxelF44) [inline include/Parser/Scene.hpp:112]:
```suggestion
             * @brief Parse the value of settings.
```

File: include/Parser/Scene.hpp [hunk 11]
```diff
@@ -84,36 +87,47 @@ class Raytracer::Scene {
...
+             * @param value to parse
+             * @return The value parsed
+             */
+            double _parseValue(const libconfig::Setting &value);
+
+            /*
+             * @brief Parse imports of scenes
```
Comment (AxelF44) [inline include/Parser/Scene.hpp:119]:
```suggestion
             * @brief Parse imports of scenes.
```

File: include/Parser/Scene.hpp [hunk 12]
```diff
@@ -84,36 +87,47 @@ class Raytracer::Scene {
...
+             * @param config of the current file
+             * @return integer 0 if no error else x
+             */
+            int _parseScenesImports(const libconfig::Config &config);
+
+            /*
+             * @brief Parse the imported scene
```
Comment (AxelF44) [inline include/Parser/Scene.hpp:126]:
```suggestion
             * @brief Parse the imported scene.
```

**Review by AxelF44 — CHANGES_REQUESTED** [review]

**Review by mathieurobert1 — COMMENTED** [review]
Nice job ! You just have to finish the user documentation.

**Review by Njord201 — CHANGES_REQUESTED** [review]
See my update on technique_documentation.pdf and utilisateur_documentation.pdf. Also rename this files in english.

**Review by AxelF44 — DISMISSED** [review]

**Review by mathieurobert1 — DISMISSED** [review]
Seems good

**Review by AxelF44 — DISMISSED** [review]

**Review by AxelF44 — DISMISSED** [review]

**Review by mathieurobert1 — DISMISSED** [review]
You have two request unresolved

**Review by AxelF44 — DISMISSED** [review]

**Review by mathieurobert1 — DISMISSED** [review]

**Review by AxelF44 — APPROVED** [review]

**Review by Njord201 — APPROVED** [review]

**Pull request merged**